packages/contracts-bedrock/test/properties/kontrol/OptimismSuperchainERC20.k.sol

```diff
@@ -0,0 +1,329 @@
+        sourceToken.sendERC20(Predeploys.CROSS_L2_INBOX, _amount, _chainId);
+
+        /* Postconditions */
+        assert(sourceToken.totalSupply() == _totalSupplyBefore - _amount);
+        assert(sourceToken.balanceOf(_sender) == _balanceBefore - _amount);
+    }
+
+    /// @custom:property-id 11
+    /// @custom:property `relayERC20` increases the token's totalSupply in the destination chain exactly by the input
+    /// amount
+    function prove_relayERC20IncreasesTotalSupply(address _from, address _to, uint256 _amount) public {
+        setUpInlined();
+
+        /* Preconditions */
+        vm.assume(_to != address(0));
+        vm.assume(notBuiltinAddress(_from));
+        vm.assume(notBuiltinAddress(_to));
+
+        uint256 _totalSupplyBefore = sourceToken.totalSupply();
+        uint256 _toBalanceBefore = sourceToken.balanceOf(_to);
+
+        vm.prank(address(MESSENGER));
+        /* Action */
+        sourceToken.relayERC20(_from, _to, _amount);
+
+        /* Postconditions */
+        assert(sourceToken.totalSupply() == _totalSupplyBefore + _amount);
+        assert(sourceToken.balanceOf(_to) == _toBalanceBefore + _amount);
+    }
+
+    /// @custom:property-id 12
+    /// @custom:property Increases the total supply on the amount minted by the bridge
+    function prove_mint(address _from, uint256 _amount) public {
+        setUpInlined();
+
+        /* Preconditions */
+        vm.assume(_from != address(0));
+        vm.assume(notBuiltinAddress(_from));
+
+        uint256 _totalSupplyBefore = sourceToken.totalSupply();
+        uint256 _balanceBefore = sourceToken.balanceOf(_from);
+
+        vm.startPrank(Predeploys.L2_STANDARD_BRIDGE);
+        /* Action */
+        sourceToken.mint(_from, _amount);
+
+        /* Postconditions */
+        assert(sourceToken.totalSupply() == _totalSupplyBefore + _amount);
+        assert(sourceToken.balanceOf(_from) == _balanceBefore + _amount);
+    }
+
+    /// @custom:property-id 13
+    /// @custom:property Supertoken total supply only decreases on the amount burned by the bridge
+    function prove_burn(address _from, uint256 _amount) public {
+        setUpInlined();
+
+        /* Preconditions */
+        vm.assume(_from != address(0));
+        vm.assume(notBuiltinAddress(_from));
+
+        vm.prank(Predeploys.L2_STANDARD_BRIDGE);
+        sourceToken.mint(_from, _amount);
+
+        uint256 _totalSupplyBefore = sourceToken.totalSupply();
+        uint256 _balanceBefore = sourceToken.balanceOf(_from);
+
+        vm.prank(Predeploys.L2_STANDARD_BRIDGE);
+        /* Action */
+        sourceToken.burn(_from, _amount);
+
+        /* Postconditions */
+        assert(sourceToken.totalSupply() == _totalSupplyBefore - _amount);
+        assert(sourceToken.balanceOf(_from) == _balanceBefore - _amount);
+    }
+
+    /// @custom:property-id 14
+    /// @custom:property Supertoken total supply starts at zero
+    function prove_totalSupplyStartsAtZero() public {
+        setUpInlined();
```

Comment:
shoudn't we us a "regular" setup for this property tho (ie deploy the contract here, with symb constructor args, instead of the state diff? I don't really see what we'd test here otherwise, this sounds like a single branch)

Reply:
I don't get which is the difference, if we already rely on the setup for the other properties, why not for this one?
The property says the total supply should start at zero, and I'm testing that after the state diff setup.
I'm open to change my mind ofc

Reply:
We use these concrete values in the state which is replayed: https://github.com/defi-wonderland/optimism/blob/602589f9fef2016c99659b0d2ca55ad82138acf3/packages/contracts-bedrock/test/properties/kontrol/KontrolBase.sol#L39

So this test is solely based on the deployment with concrete value, and only assess one single case -> I agree it shouldn't change much of the underlying logic, but if we want to test the initial state, shouldn't these concrete values be symbolic instead (ie catching a bug where total supply > 0 if decimals is 10**123)?